packages/vscode-extension/src/webview/views/EnvEditor.tsx

```diff
@@ -0,0 +1,163 @@
+import { useState } from "react";
+import {
+  VscodeTable,
+  VscodeTableHeader,
+  VscodeTableHeaderCell,
+  VscodeTableBody,
+  VscodeTableRow,
+  VscodeTableCell,
+  VscodeTextfield,
+  VscodeButton as Button,
+} from "@vscode-elements/react-elements";
+import ToolbarButton from "../components/shared/VscodeToolbarButton";
+import "./EnvEditor.css";
+
+interface EnvEditorProps {
+  initialValue?: Record<string, string>;
+  onChange?: (env: Record<string, string>) => void;
+}
+
+function EnvEditor({ initialValue, onChange }: EnvEditorProps) {
+  const [env, setEnv] = useState<Record<string, string>>(initialValue ?? {});
+  const [editingKey, setEditingKey] = useState<string | null>(null);
+  const [editingKeyValue, setEditingKeyValue] = useState<string>("");
+  const [editingValue, setEditingValue] = useState<string>("");
+  const [showAddForm, setShowAddForm] = useState<boolean>(false);
+
+  const updateEnv = (newEnv: Record<string, string>) => {
+    setEnv(newEnv);
+    onChange?.(newEnv);
+  };
+
+  const startEditing = (key: string | null) => {
```

Comment:
Could we have this focus the `key` text field for the new/edited entry as well?

Reply:
I wanted to do that but for some reason autofocus prop doesn't work in vscode-elements. Would need to add a bunch of more workarounds for this to work apparently.